tests/test_discovery.py

```diff
@@ -78,44 +78,51 @@
 @wallswitch_iot
 async def test_type_detection_switch(dev: Device):
     d = Discover._get_device_class(dev._last_update)("localhost")
-    assert d.is_wallswitch
-    assert d.device_type == DeviceType.WallSwitch
+    with pytest.deprecated_call(match="use device_type property instead"):
+        assert d.is_wallswitch
+    assert d.device_type is DeviceType.WallSwitch
 
 
 @plug_iot
 async def test_type_detection_plug(dev: Device):
     d = Discover._get_device_class(dev._last_update)("localhost")
-    assert d.is_plug
+    with pytest.deprecated_call(match="use device_type property instead"):
+        assert d.is_plug
```

Comment:
```suggestion
```
I think the correct thing to do here is just drop the `is_plug` check altogether. However if we adopt the more generic approach of device type categorization we're doing for `smartcamera` we'd probably change the way we do these tests and pass in all the fixtures with device_type `plug` and check they're a member of some hardcoded list (not for this PR)